File: 백준 문제 풀이 모음/class 3/1931.py
import sys

x = int(sys.stdin.readline())
time = []

for i in range(x) :
    time.append(list(map(int, input().split())))

time = sorted(time, key=lambda a: a[0])
time = sorted(time, key=lambda a: a[1])

last = 0
count = 0

for i, j in time :
    if i >= last :
        count += 1
        last = j

print(count)

File: 백준 문제 풀이 모음/kotlin/class 3/1931.kt
package class3

fun main() = with(System.`in`.bufferedReader()) {

    val n = readLine().toInt()
    val arr = MutableList<Pair<Int, Int>>(0) { Pair(0,0)}

    repeat(n) {
        val str = readLine().split(' ').toList()
        val a = str[0].toInt()
        val b = str[1].toInt()

        arr.add(Pair(a,b))
    }


    arr.sortBy {
        it.first
    }
    arr.sortBy {
        it.second
    }
//    arr.forEach {
//        print(it.first)
//        print(' ')
//        print(it.second)
//        println()
//    }


    var count = 0
    var now = 0

    arr.forEach {
        if (it.first >= now) {
            count ++
            now = it.second
        }
    }



    println(count)
}